### PR TITLE
Theme and responsive design

### PR DESCRIPTION
**Pull request by akash-1318**
## Implemented responsive design and light/dark mode - 
I have implemented responsive design and light/dark mode in my video library also implemented some additional features like search videos and loader.

## Files to check - 
1. src/contexts/theme-context.js
2. src/style/responsive.css
3. src/utils/filter-videos.js
4. src/pages/single-video.js/single-video.js

https://user-images.githubusercontent.com/64393554/168666982-26ab4191-faf2-43e8-a587-0c4dcfc49ef4.mov

### CODE REVIEW

File: src/components/modal/modal.js [hunk 1]
```diff
@@ -55,7 +57,7 @@ function Modal() {
   return (
     <>
       <div className="modal__container">
-        <div className="modal">
+        <div className={`modal ${theme === "light" ? "" : "dark"}`}>
           <div className="modal__header">
             <p onClick={() => console.log(currentVideo)}>Save to...</p>
             <i class="bx bx-x" onClick={() => setModal(!modal)}></i>
```
Comment (shashikantmaurya0007) [inline src/components/modal/modal.js:63]: try to keep jsx and logic seperate

File: src/pages/playlist/playlist.js [hunk 2]
```diff
@@ -19,15 +19,6 @@ function Playlist() {
...
           <h1 className="page__text">Playlists</h1>
-          {playlists.length > 0 && (
-            <button
-              className="clear__history"
-            //   onClick={() => clearAllHistory(dispatch, authToken)}
-            >
-              {" "}
-              <i class="bx bx-trash"></i> Delete all Playlists
-            </button>
-          )}
```
Comment (shashikantmaurya0007) [inline src/pages/playlist/playlist.js:30]: remove commented code

File: src/pages/single-video.js/single-video.js [hunk 3]
```diff
@@ -29,13 +30,8 @@ function SingleVideo() {
...
+        setLoader(false)
         setSingleVideo(resp.data.video);
-        setRelatedVideos(
-          videosData.filter(
-            (video) => video.category === resp.data.video.category
-          )
-        );
       } catch (err) {
         console.log(err);
       }
```
Comment (shashikantmaurya0007) [inline src/pages/single-video.js/single-video.js:37]: you can make custom hook for this to make code look more clean

**Review by shashikantmaurya0007 — APPROVED** [review]
I have added few comment .rest code looks good .liked that toggle feature for sidebar

**Review by chanchal-panpaliya — APPROVED** [review]
1. code looks clean with proper naming 
2. good UI

**Pull request merged**